content/blog/cloud-native-secret-management-with-pulumi-esc-and-external-secrets-operator/index.md

```diff
@@ -0,0 +1,450 @@
+                project: "eso-esc-demo",
+                environment: "esp-esc-app",
+                accessToken: {
+                    secretRef: {
+                        name: mySecret.metadata.name,
+                        key: "PULUMI_ACCESS_TOKEN",
+                        namespace: mySecret.metadata.namespace,
+                    },
+                },
+            },
+        },
+    },
+});
+```
+
+We can check that the secret store was successfully synchronized by running:
+
+```bash
+kubectl get clustersecretstore secret-store -o jsonpath='{.status.conditions[?(@.type=="Ready")].status}'
+```
+
+### Step 3: Deploy an Application and Reference the Secret
+
+Now, we can deploy an application that references the secret from the Kubernetes cluster. I am going to deploy the app
+called `PodInfo` which is a simple application that exposes the pod information as an API. Add the following code to the
+`index.ts`:
+
+```typescript
+import * as k8s from "@pulumi/kubernetes";
+import * as pulumi from "@pulumi/pulumi";
+
+// cut for brevity
+const podInfo = new k8s.helm.v3.Release("podinfo", {
+    chart: "podinfo",
+    version: "6.7.0",
+    namespace: "podinfo",
+    createNamespace: true,
+    repositoryOpts: {
+        repo: "https://stefanprodan.github.io/podinfo",
+    },
+    values: {
+        extraEnvs: [
+            {
+                name: "FROM_ESC_VIA_ESO",
+                valueFrom: {
+                    secretKeyRef: {
+                        name: "esc-secret-store",
+                        key: "hello-secret",
+                    }
+                }
+            },
+            {
+                name: "FROM_ESC_VIA_ESO_2",
+                valueFrom: {
+                    secretKeyRef: {
+                        name: "esc-secret-store",
+                        key: "hello",
+                    }
+                }
+            }
+        ]
+    },
+});
+
+const externalSecretPodInfo = new k8s.apiextensions.CustomResource("external-secret-podinfo", {
+    apiVersion: "external-secrets.io/v1beta1",
+    kind: "ExternalSecret",
+    metadata: {
+        name: "esc-secret-store",
+        namespace: podInfo.namespace,
+    },
+    spec: {
+        dataFrom: [
+            {
+                extract: {
+                    conversionStrategy: "Default",
+                    key: "app",
+                }
+            }
+        ],
+        refreshInterval: "10s",
+        secretStoreRef: {
+            kind: clusterSecretStore.kind,
+            name: clusterSecretStore.metadata.name,
+        }
+    },
+});
+```
+
+This will deploy the `PodInfo` application and create an `ExternalSecret` resource that references the secret from the
+Pulumi ESC project. We mount the secret into the pod as an environment variable by using the `extraEnvs` field in the
+Helm chart.
+
+{{% notes type="info" %}}
+
+In your real-world application, you could add different ways to reference the secret, like mounting it as a volume or
```

Comment:
...->"In a real-world application, you might use alternate ways to reference the secret, like mounting it as a volume..."